newton/_src/utils/schema_resolver.py

```diff
@@ -0,0 +1,479 @@
+        for _prim_type, var_map in mapping_items:
+            try:
+                var_items = var_map.items()
+            except AttributeError:
+                continue
+            for _var, specs in var_items:
+                for spec in specs:
+                    names.add(spec.usd_name)
+        self._engine_attributes: list[str] = list(names)
+
+    @property
+    def engine_attr_prefix(self) -> str:
+        """Return the main engine attribute prefix (e.g., 'newton:', 'physx', 'mjc:')"""
+        return f"{self.name}:"
+
+    def get_value(self, prim, prim_type: str, key: str) -> tuple[Any, str] | None:
+        """
+        Get attribute value for a given prim type and key.
+
+        Args:
+            prim: USD prim to query
+            prim_type: Prim type ("scene", "joint", "shape", "body", "material", "actuator")
+            key: Attribute key within the prim type
+
+        Returns:
+            Tuple of (value, usd_attr_name) if found, None otherwise
+        """
+        if prim is None:
+            return None
+        for spec in self.mapping.get(prim_type, {}).get(key, []):
+            v = _get_attr(prim, spec.usd_name)
+            if v is not None:
+                return (spec.transform(v) if spec.transform is not None else v), spec.usd_name
+        return None
+
+    def collect_prim_engine_attrs(self, prim) -> dict[str, Any]:
+        """
+        Collect engine-specific attributes for a single prim.
+        Returns dictionary of engine-specific attributes for this prim.
+        """
+        if prim is None:
+            return {}
+
+        # Collect explicit attribute names defined in the plugin mapping (precomputed)
+        prim_engine_attrs = (
+            _collect_engine_mapped_attrs(prim, self._engine_attributes) if self._engine_attributes else {}
+        )
+
+        # Collect attributes by known engine-specific prefixes
+        all_prefixes = [self.engine_attr_prefix]
+        if self.extra_attr_prefixes:
+            all_prefixes.extend(self.extra_attr_prefixes)
+        prefixed_attrs = _collect_engine_specific_attrs(prim, all_prefixes)
+
+        # Merge and return (explicit names take precedence)
+        merged: dict[str, Any] = {}
+        merged.update(prefixed_attrs)
+        merged.update(prim_engine_attrs)
+        return merged
+
+
+def _get_attr(prim, name: str):
+    if prim is None:
+        return None
+    attr = prim.GetAttribute(name)
+    if not (attr and attr.IsValid() and attr.HasAuthoredValue()):
+        return None
+    return attr.Get()
+
+
+def _collect_engine_mapped_attrs(prim, names: list[str]) -> dict[str, Any]:
+    """Collect engine-specific attributes authored on the prim that have direct mappings in the plugin mapping"""
+    out = {}
+    for n in names:
+        v = _get_attr(prim, n)
+        if v is not None:
+            out[n] = v
+    return out
+
+
+def _collect_engine_specific_attrs(prim, prefixes: list[str]) -> dict[str, Any]:
```

Comment:
It should be possible to simplify this function using `prim.GetAuthoredPropertiesInNamespace(prefixes)`

There is an overload that accepts a list/vector, if you need to pass in all the `physx*` namespaces